File: Beecrowd_Parte2/Echianos_2593.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Leitura do texto
        String texto = scanner.nextLine();
        
        // Leitura da quantidade de palavras
        int qtdPalavras = scanner.nextInt();
        scanner.nextLine(); // Consumir quebra de linha
        
        // Leitura das palavras
        String[] palavras = scanner.nextLine().split(" ");
        
        // Processamento das palavras
        for (String palavra : palavras) {
            boolean primeiraPosicao = true;
            boolean encontrou = false;
            
            // Procurar a palavra no texto
            int posInicial = 0;
            while ((posInicial = texto.indexOf(palavra, posInicial)) != -1) {
                if (!primeiraPosicao) {
                    System.out.print(" ");
                }
                System.out.print(posInicial);
                primeiraPosicao = false;
                encontrou = true;
                
                // Avançar para a próxima posição após a última ocorrência
                posInicial += palavra.length();
            }
            
            // Se não encontrou nenhuma ocorrência da palavra
            if (!encontrou) {
                System.out.print("-1");
            }
            
            System.out.println(); // Nova linha para a próxima palavra
        }
        
        scanner.close();
    }
}

File: Beecrowd_Parte2/Echianos_2593.c
#include <stdio.h>
#include <string.h>

int main() {
    char texto[10001]; // Considerando o texto com no máximo 10.000 caracteres
    char palavras[128][51]; // Array de palavras com no máximo 128 palavras de até 50 caracteres cada
    
    // Leitura do texto
    fgets(texto, sizeof(texto), stdin);
    texto[strcspn(texto, "\n")] = '\0'; // Remove o newline lido por fgets
    
    // Leitura da quantidade de palavras
    int qtdPalavras;
    scanf("%d", &qtdPalavras);
    getchar(); // Consumir o caractere newline após o número de palavras
    
    // Leitura das palavras
    for (int i = 0; i < qtdPalavras; i++) {
        scanf("%s", palavras[i]);
    }
    
    // Processamento das palavras
    for (int i = 0; i < qtdPalavras; i++) {
        char *posicao = strstr(texto, palavras[i]);
        if (posicao != NULL) {
            // Imprimir a primeira posição encontrada
            while (posicao != NULL) {
                int pos = posicao - texto;
                printf("%d ", pos);
                posicao = strstr(posicao + 1, palavras[i]);
            }
        } else {
            // Se a palavra não foi encontrada, imprimir -1
            printf("-1");
        }
        printf("\n");
    }
    
    return 0;
}
